agent/config/config_test.go

```diff
@@ -211,170 +188,146 @@ func TestInvalideValueDockerStopTimeout(t *testing.T) {
-	}
+	assert.NoError(t, err)
+	assert.True(t, cfg.TaskIAMRoleEnabled, "Wrong value for TaskIAMRoleEnabled")
 }
 
 func TestTaskIAMRoleForHostNetworkEnabled(t *testing.T) {
+	defer os.Clearenv()
+	setupRegion()
 	os.Setenv("ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST", "true")
 	cfg, err := NewConfig(ec2.NewBlackholeEC2MetadataClient())
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if !cfg.TaskIAMRoleEnabledForNetworkHost {
-		t.Errorf("Wrong value for TaskIAMRoleEnabledForNetworkHost: %v", cfg.TaskIAMRoleEnabledForNetworkHost)
-	}
+	assert.NoError(t, err)
+	assert.True(t, cfg.TaskIAMRoleEnabledForNetworkHost, "Wrong value for TaskIAMRoleEnabledForNetworkHost")
 }
 
 func TestCredentialsAuditLogFile(t *testing.T) {
+	defer os.Clearenv()
+	setupRegion()
 	dummyLocation := "/foo/bar.log"
 	os.Setenv("ECS_AUDIT_LOGFILE", dummyLocation)
 	cfg, err := NewConfig(ec2.NewBlackholeEC2MetadataClient())
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if cfg.CredentialsAuditLogFile != dummyLocation {
-		t.Errorf("Wrong value for CredentialsAuditLogFile: %v", cfg.CredentialsAuditLogFile)
-	}
+	assert.NoError(t, err)
+	assert.Equal(t, cfg.CredentialsAuditLogFile, dummyLocation, "Wrong value for CredentialsAuditLogFile")
 }
 
 func TestCredentialsAuditLogDisabled(t *testing.T) {
+	defer os.Clearenv()
+	setupRegion()
 	os.Setenv("ECS_AUDIT_LOGFILE_DISABLED", "true")
 	cfg, err := NewConfig(ec2.NewBlackholeEC2MetadataClient())
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if !cfg.CredentialsAuditLogDisabled {
-		t.Errorf("Wrong value for CredentialsAuditLogDisabled: %v", cfg.CredentialsAuditLogDisabled)
-	}
+	assert.NoError(t, err)
+	assert.True(t, cfg.CredentialsAuditLogDisabled, "Wrong value for CredentialsAuditLogDisabled")
 }
 
 func TestImageCleanupMinimumInterval(t *testing.T) {
+	defer os.Clearenv()
+	setupRegion()
 	os.Setenv("ECS_IMAGE_CLEANUP_INTERVAL", "1m")
 	cfg, err := NewConfig(ec2.NewBlackholeEC2MetadataClient())
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if cfg.ImageCleanupInterval != DefaultImageCleanupTimeInterval {
-		t.Errorf("Wrong value for ImageCleanupInterval: %v", cfg.ImageCleanupInterval)
-	}
+	assert.NoError(t, err)
+	assert.Equal(t, cfg.ImageCleanupInterval, DefaultImageCleanupTimeInterval, "Wrong value for ImageCleanupInterval")
 }
 
 func TestImageCleanupMinimumNumImagesToDeletePerCycle(t *testing.T) {
+	defer os.Clearenv()
+	setupRegion()
 	os.Setenv("ECS_NUM_IMAGES_DELETE_PER_CYCLE", "-1")
 	cfg, err := NewConfig(ec2.NewBlackholeEC2MetadataClient())
-	if err != nil {
-		t.Fatal(err)
-	}
+	assert.NoError(t, err)
+	assert.Equal(t, cfg.NumImagesToDeletePerCycle, DefaultNumImagesToDeletePerCycle, "Wrong value for NumImagesToDeletePerCycle")
+}
 
-	if cfg.NumImagesToDeletePerCycle != DefaultNumImagesToDeletePerCycle {
-		t.Errorf("Wrong value for NumImagesToDeletePerCycle: %v", cfg.NumImagesToDeletePerCycle)
-	}
+func TestTaskResourceLimitsOverride(t *testing.T) {
+	defer os.Clearenv()
```

Comment:
I'm not a fan of all the `os.Clearenv` directives here. Can we reduce the scope of the cleanup to be just 
```
defer os.Unsetenv("SOME_SPECIFIC_THING_1")
defer os.Unsetenv("SOME_SPECIFIC_THING_2_IF_NEEDED")
``` 
? My main concern is that it might have some unintended consequences for other unit tests.